src/Client/BuzzHouse/Generator/QueryOracle.h

```diff
@@ -21,9 +22,12 @@ class QueryOracle
     void findTablesWithPeersAndReplace(RandomGenerator & rg, google::protobuf::Message & mes, StatementGenerator & gen);
 
 public:
-    explicit QueryOracle(const FuzzConfig & ffc) : fc(ffc) { buf.reserve(4096); }
+    explicit QueryOracle(const FuzzConfig & ffc) : fc(ffc), qfile(ffc.db_file_path / "query.data") { buf.reserve(4096); }
 
-    int processOracleQueryResult(bool first, bool success, const std::string & oracle_name);
+    int ResetOracleValues();
+    int SetIntermediateStepSuccess(bool success);
+    int processFirstOracleQueryResult(bool success);
```

Comment:
I don't understand what's the point of all these methods returning `int` if they always return 0. That's why you need to do nasty stuff to ignore the returned value.